filestore/core.py

```diff
@@ -0,0 +1,88 @@
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
+
+import six
+from document import Document
+from jsonschema import validate as js_validate
+from bson import ObjectId
+
+
+class DatumNotFound(Exception):
+    pass
+
+
+def get_datum(col, eid, _DATUM_CACHE, get_spec_handler, logger):
```

Comment:
Structurally, I think this would make more sense as a method on `FileStoreRO`. On the other hand, this functional implementation might make it possible to do some neat variations -- `_DATUM_CACHE` might not be a `dict` for example.

For easier comprehension, can you not name `_DATUM_CACHE` like a private constant variable?